lib/semian.rb

```diff
@@ -157,7 +158,8 @@ def register(name, **options)
       raise ArgumentError, 'Both bulkhead and circuitbreaker cannot be disabled.'
     end
 
-    resources[name] = ProtectedResource.new(name, bulkhead, circuit_breaker)
+    resources.set(name, ProtectedResource.new(name, bulkhead, circuit_breaker))
+    resources.get(name).value
```

Comment:
Doing a set and then a get sounds like you're doing extra unnecessary book-keeping work.